internal/worker/bootstrap/worker.go

```diff
@@ -215,6 +234,42 @@ func (w *bootstrapWorker) reportInternalState(state string) {
 	}
 }
 
+// initAPIHostPorts sets the initial API host/port addresses in state.
+func (w *bootstrapWorker) initAPIHostPorts(ctx context.Context, controllerConfig controller.Config, pAddrs network.ProviderAddresses, apiPort int) error {
+	addrs, err := w.providerAddressesToSpaceAddresses(ctx, pAddrs)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	hostPorts := []network.SpaceHostPorts{network.SpaceAddressesWithPort(addrs, apiPort)}
+	hostPortsForAgents, err := w.cfg.SpaceService.FilterHostPortsForManagementSpace(ctx, controllerConfig, hostPorts)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	return w.cfg.SystemState.SetAPIHostPorts(controllerConfig, hostPorts, hostPortsForAgents)
+}
+
+func (w *bootstrapWorker) providerAddressesToSpaceAddresses(ctx context.Context, providerAddresses network.ProviderAddresses) (network.SpaceAddresses, error) {
+	addrs := make(network.SpaceAddresses, len(providerAddresses))
+
+	for i, pa := range providerAddresses {
+		addrs[i] = network.SpaceAddress{MachineAddress: pa.MachineAddress}
+		if pa.SpaceName != "" {
+			spInfo, err := w.cfg.SpaceService.SpaceByName(ctx, string(pa.SpaceName))
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			if spInfo == nil {
+				return nil, errors.NotFoundf("space with name %q", pa.SpaceName)
+			}
+			addrs[i].SpaceID = spInfo.ID
+		}
+	}
+
+	return addrs, nil
+}
```

Comment:
test me

Reply:
so, this has been moved, finally I can use the `ToSpaceAddresses()` method of the ProviderAddresses, which simplifies things